packages/client/src/components/Component.svelte

```diff
@@ -283,10 +283,23 @@
         const dependsOnKey = setting.dependsOn.setting || setting.dependsOn
         const dependsOnValue = setting.dependsOn.value
         const realDependentValue = instance[dependsOnKey]
+
+        const sectionDependsOnKey =
+          setting.sectionDependsOn?.setting || setting.sectionDependsOn
```

Comment:
I see what you're doing with this (I remember the awkwardness before of trying to define the chart settings in a way that the "depends on" stuff worked with only a single key) but to be honest this would motivate me to rewrite dependsOn to just support an array of keys. We could still do what you're doing below and dynamically enrich the setting dependsOn keys with any keys from the section itself.

We can handle that separately though - this PR is big enough already!